packages/eslint-plugin-template/src/rules/attributes-order.ts

```diff
@@ -328,23 +334,28 @@ function extractTemplateAttrs(
 
   const attrs = node.parent.templateAttrs.map(toStructuralDirectiveOrderType);
 
-  // Pick up on any subsequent `let` bindings, e.g. `index as i`
-  let sourceEnd = attrs[attrs.length - 1].sourceSpan.end;
-  node.parent.variables.forEach((v) => {
-    if (
-      v.sourceSpan.start.offset <= sourceEnd.offset &&
-      sourceEnd.offset < v.sourceSpan.end.offset
-    ) {
-      sourceEnd = v.sourceSpan.end;
+  let keyEnd = attrs[0].keySpan?.end;
+  if (keyEnd?.getContext(0, 0)?.after === '=') {
+    keyEnd = keyEnd.moveBy(1);
+    const apos = keyEnd.getContext(0, 0)?.after;
+    if (apos === "'" || apos === '"') {
+      do {
+        keyEnd = keyEnd.moveBy(1);
+      } while (keyEnd.getContext(0, 0)?.after !== apos);
+    } else {
+      while (!/[\s>]/.test(keyEnd.getContext(0, 0)?.after ?? '')) {
+        keyEnd = keyEnd.moveBy(1);
+      }
     }
```

Comment:
Instead of trying to merge multiple spans it works out the value span separately. This deals with contained template variables and extra whitespace within the value.